### PR TITLE
Fix to text shrinking

### PR DESCRIPTION
**Pull request by rcancro**
Summary: Previously the first thing we did when calculating the size of text was to truncate it. This could lead to ASFontSizeAdjuster being sent a truncated string which, obviously, never needed to shrink. If we change the order then we first shrink, then truncate if still necessary.

### CODE REVIEW

File: AsyncDisplayKit/TextKit/ASTextKitRenderer.mm [hunk 1]
```diff
@@ -137,25 +137,30 @@ - (void)setConstrainedSize:(CGSize)constrainedSize
...
       [ASTextKitFontSizeAdjuster adjustFontSizeForAttributeString:scaledString withScaleFactor:_currentScaleFactor];
       scaledTextStorage = [[NSTextStorage alloc] initWithAttributedString:scaledString];
       
       [textStorage removeLayoutManager:layoutManager];
       [scaledTextStorage addLayoutManager:layoutManager];
-    }
+    }];
+  }
+  
+  [self truncater];
```
Comment (garrettmoon) [inline AsyncDisplayKit/TextKit/ASTextKitRenderer.mm]: Should we rename this method to truncateIfNecessary? Something better even? It's weird that calling [self truncater] does something other than return a truncater.

Comment (rcancro) [inline AsyncDisplayKit/TextKit/ASTextKitRenderer.mm]: I don't disagree.

File: AsyncDisplayKit/TextKit/ASTextKitTruncating.h [hunk 2]
```diff
@@ -33,4 +33,9 @@
      truncationAttributedString:(NSAttributedString *)truncationAttributedString
          avoidTailTruncationSet:(NSCharacterSet *)avoidTailTruncationSet;
 
+/**
+ *  Actually do the truncation.
```
Comment (levi) [inline AsyncDisplayKit/TextKit/ASTextKitTruncating.h]: LOL. Might want some more context for this note.

**Comment by levi** [comment]
Looks good.

File: AsyncDisplayKit/TextKit/ASTextKitTruncating.h (hunk 2 above)
Comment (appleguy) [inline AsyncDisplayKit/TextKit/ASTextKitTruncating.h]: Haha, yeah - let's leave out unnecessary comments, though humor is welcome in ones that also provide information to the reader :)

**Pull request merged**